app/models/sipity/workflow.rb

```diff
@@ -39,20 +39,19 @@ def self.find_active_workflow_for(admin_set_id:)
     # Within the given permission_template scope:
     #   * Activate the specified workflow_id or workflow_name
     #   * Deactivate the other workflows
-    #
-    # TODO: Resolve https://github.com/samvera/hyrax/issues/2151 as the documentation is aspirational and not reality
-    #
     # @param permission_template [Hyrax::PermissionTemplate] The scope for activation of the workflow id
     # @param workflow_id [Integer] The workflow_id within the given permission_template that should be activated
     # @param workflow_name [String] The name of the workflow within the given permission template that should be activated
-    # @return [TrueClass]
+    # @return [Sipity::Workflow] active workflow
     # @raise [ActiveRecord::RecordNotFound] When we have a mismatch on permission template and workflow id or workflow name
     # @raise [RuntimeError] When you don't specify a workflow_id or workflow_name
     def self.activate!(permission_template:, workflow_id: nil, workflow_name: nil)
       raise "You must specify a workflow_id or workflow_name to activate!" if workflow_id.blank? && workflow_name.blank?
-      finder_attributes = { permission_template: permission_template, id: workflow_id, name: workflow_name }.compact
-      Sipity::Workflow.find_by!(finder_attributes).tap do |workflow|
-        Sipity::Workflow.where(permission_template: permission_template, active: true).update(active: nil)
+      workflow_to_activate = Sipity::Workflow.find_by!({ permission_template: permission_template, id: workflow_id, name: workflow_name }.compact)
+      active_workflow = Sipity::Workflow.where(permission_template: permission_template, active: true)
+      return workflow_to_activate if workflow_to_activate == active_workflow.first
+      workflow_to_activate.tap do |workflow|
+        active_workflow.update(active: nil)
```

Comment:
Won't `active_workflow` be multiple `Sipity::Workflow` objects here?

Reply:
No, there is always only one active workflow, though I'm not positive right now where that gets enforced.